### PR TITLE
Fix bug in methods that failed when `predict` was called before the first partial_fit/fit

### LINKED ISSUE #158: [BUG] Online Boosting crashes when using KNN as base learner
If in the first call to partial_fit in Onlineboosting with base estimator KNNADWIN I got this Error:
ValueError: KNN must be (partially) fitted on n_neighbors samples before doing any prediction.
How can I solve that?
Thanks

### PR DESCRIPTION
**Pull request by jacobmontiel**
Stream estimators shall be able to return predictions at any time. Some methods were failing when predictions were requested and the model was empty. This PR addresses this issue for:
* `KNNClassifier`
* `OnlineAdaC2Classifier`
* `OnlineBoosting`
* `OnlineUnderOverBaggingClassifier`
* `OnlineSMOTEBaggingClassifier`
* `OnlineCSB2Classifier`

Changes proposed in this pull request:

- Return a default value (zero) for predictions if the model is empty.
- Update documentation of the KNNClassifier.
- Handle RuntimeWarning from numpy operations in `Online****`
- Fix typo in example code for `MonteCarloClassifierChain`
- Fixes #158

---

Checklist

- [x] Code complies with PEP-8 and is consistent with the framework.
- [x] Code is properly documented.
- [x] Tests are included for new functionality or updated accordingly.
- [x] Travis CI build passes with no errors.
- [ ] Test Coverage is maintained (threshold is -0.2%).
- [x] Files changed (update, add, delete) are in the PR's scope (no extra files are included).

### CODE REVIEW

**Comment by jacobmontiel** [comment]
Note: Changes to `VeryFastDecisionRulesClassifier` are from the development branch (upstream).

File: src/skmultiflow/meta/online_boosting.py [hunk 1]
```diff
@@ -109,6 +108,7 @@ def __configure(self):
         self.lam_sc = np.zeros(self.actual_n_estimators)
         self.lam_sw = np.zeros(self.actual_n_estimators)
         self.epsilon = np.zeros(self.actual_n_estimators)
+        self._is_empty = True
```
Comment (smastelini) [inline src/skmultiflow/meta/online_boosting.py]: Is this variable used in other contexts?

File: src/skmultiflow/meta/online_smote_bagging.py [hunk 2]
```diff
@@ -144,6 +143,9 @@ def partial_fit(self, X, y, classes=None, sample_weight=None):
         self
 
         """
+        if self.adwin_ensemble is None:
```
Comment (smastelini) [inline src/skmultiflow/meta/online_smote_bagging.py]: Should not `ensemble` be used here, instead? (to cope with other ensemble models)

File: src/skmultiflow/meta/online_smote_bagging.py [hunk 3]
```diff
@@ -285,27 +287,37 @@ def predict_proba(self, X):
...
-                for n in range(r):
-                    for l in range(len(partial_proba[n])):
-                        try:
-                            proba[n][l] += partial_proba[n][l]
-                        except IndexError:
-                            proba[n].append(partial_proba[n][l])
-        except ValueError:
-            return np.zeros((r, 1))
-        except TypeError:
+        if self.adwin_ensemble is None:
```
Comment (smastelini) [inline src/skmultiflow/meta/online_smote_bagging.py]: Should not `ensemble` be used here, instead? (to cope with other ensemble models)

File: src/skmultiflow/meta/online_under_over_bagging.py [hunk 4]
```diff
@@ -147,6 +145,9 @@ def partial_fit(self, X, y, classes=None, sample_weight=None):
         passed in the first partial_fit call, or if they are passed in further
         calls but differ from the initial classes list passed..
         """
+        if self.adwin_ensemble is None:
```
Comment (smastelini) [inline src/skmultiflow/meta/online_under_over_bagging.py]: Should not `self.ensemble` be used here, instead? (to cope with other ensemble models)

File: src/skmultiflow/meta/online_under_over_bagging.py [hunk 5]
```diff
@@ -263,27 +264,37 @@ def predict_proba(self, X):
...
-                for n in range(r):
-                    for l in range(len(partial_proba[n])):
-                        try:
-                            proba[n][l] += partial_proba[n][l]
-                        except IndexError:
-                            proba[n].append(partial_proba[n][l])
-        except ValueError:
-            return np.zeros((r, 1))
-        except TypeError:
+        if self.adwin_ensemble is None:
```
Comment (smastelini) [inline src/skmultiflow/meta/online_under_over_bagging.py]: Should not `self.ensemble` be used here, instead? (to cope with other ensemble models)

File: src/skmultiflow/lazy/knn_classifier.py [hunk 6]
```diff
@@ -185,21 +156,12 @@ def predict(self, X):
         return np.array(predictions)
 
     def predict_proba(self, X):
-        """ predict_proba
-         
-        Calculates the probability of each sample in X belonging to each 
-        of the labels, based on the knn algorithm.
+        """ Estimates the probability of each sample in X belonging to each of the class-labels.pyte
```
[the remark this answers is not in the document]
Comment (jacobmontiel) [reply src/skmultiflow/lazy/knn_classifier.py]: Fixed in 0d7aaeeb2841b9e5dbfe80fb821691ee2fbbd166

File: src/skmultiflow/meta/classifier_chains.py [hunk 7]
```diff
@@ -45,7 +45,7 @@ class ClassifierChain(BaseSKMObject, ClassifierMixin, MetaEstimatorMixin, MultiO
     >>> print(cc.predict(X))
     >>>
     >>> print("MCC")
-    >>> mcc = MCC(SGDClassifier(max_iter=100, loss='log', random_state=1), M=1000)
+    >>> mcc = MonteCarloClassifierChain(base_estimator=SGDClassifier(max_iter=100, loss='log', random_state=1), M=1000)
```
[the remark this answers is not in the document]
Comment (jacobmontiel) [reply src/skmultiflow/meta/classifier_chains.py]: Updated PR documentation

File: src/skmultiflow/meta/online_boosting.py (hunk 1 above)
Comment (jacobmontiel) [reply src/skmultiflow/meta/online_boosting.py]: No, it should not be there. Removed in 0d7aaeeb2841b9e5dbfe80fb821691ee2fbbd166

File: src/skmultiflow/meta/online_smote_bagging.py (hunk 2 above)
Comment (jacobmontiel) [reply src/skmultiflow/meta/online_smote_bagging.py]: Correct, fixed in 0d7aaeeb2841b9e5dbfe80fb821691ee2fbbd166

File: src/skmultiflow/meta/online_smote_bagging.py (hunk 3 above)
Comment (jacobmontiel) [reply src/skmultiflow/meta/online_smote_bagging.py]: Correct, fixed in 0d7aaeeb2841b9e5dbfe80fb821691ee2fbbd166

File: src/skmultiflow/meta/online_under_over_bagging.py (hunk 4 above)
Comment (jacobmontiel) [reply src/skmultiflow/meta/online_under_over_bagging.py]: Correct, fixed in 0d7aaeeb2841b9e5dbfe80fb821691ee2fbbd166

File: src/skmultiflow/meta/online_under_over_bagging.py (hunk 5 above)
Comment (jacobmontiel) [reply src/skmultiflow/meta/online_under_over_bagging.py]: Correct, fixed in 0d7aaeeb2841b9e5dbfe80fb821691ee2fbbd166

**Pull request merged**